backend/src/ee/services/access-approval-request/access-approval-request-dal.ts

```diff
@@ -11,6 +11,32 @@ export type TAccessApprovalRequestDALFactory = ReturnType<typeof accessApprovalR
 
 export const accessApprovalRequestDALFactory = (db: TDbClient) => {
   const accessApprovalRequestOrm = ormify(db, TableName.AccessApprovalRequest);
+  const projectUserAdditionalPrivilegeOrm = ormify(db, TableName.ProjectUserAdditionalPrivilege);
+
+  const deleteMany = async (filter: TFindFilter<TAccessApprovalRequests>, tx?: Knex) => {
+    try {
+      const handleDeletion = async (processedTx: Knex) => {
```

Comment:
You could do a  `const processedTx = tx || (await db.transaction())` and used it like normal without function inside function pattern.

https://knexjs.org/guide/transactions.html